Album/ViewControllers/ProfileControllers/CloseFriendViewController.m

```diff
@@ -8,134 +8,146 @@
 
 @property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (weak, nonatomic) IBOutlet UISearchBar *closeSearchBar;
 @property (strong, nonatomic) NSArray *friendsArr;
 @property (strong, nonatomic) NSArray *friendshipArr;
 @property (strong, nonatomic) NSArray *otherFriendshipArr;
-@property (strong, nonatomic) ParseAPIHelper *apiHelper;
-
+@property (strong, nonatomic) NSArray *filterFriendsArr;
+@property (strong, nonatomic) NSArray *filterFriendshipArr;
+@property (strong, nonatomic) NSArray *filterOtherFriendshipArr;
 @end
 
 @implementation CloseFriendViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    // Set API helper
-    self.apiHelper = [[ParseAPIHelper alloc] init];
-    // Assign collection view delegate and dataSource
+    // Assign delegate and dataSource
     self.tableView.delegate = self;
     self.tableView.dataSource = self;
+    self.closeSearchBar.delegate = self;
+    [self setFriendshipStructure];
+}
+
+- (void)setFriendshipStructure {
     // Fetch friends
-    PFQuery *friendQuery = [PFQuery queryWithClassName:classNameFriendship];
-    NSLog(@"%@",self.user.objectId);
-    [friendQuery whereKey:@"requesterId" equalTo:self.user.objectId];
-    [friendQuery whereKey:@"hasFriended" equalTo:@(2)];
-    [friendQuery findObjectsInBackgroundWithBlock:^(NSArray *friendships, NSError *error) {
-        self.friendshipArr = friendships;
-        NSMutableArray *friendArr = [[NSMutableArray alloc] init];
-        NSMutableArray *otherFriendships = [[NSMutableArray alloc] init];
-        if (friendships != nil) {
-            for (Friendship *friendship in friendships) {
-                NSString *friendId = friendship[@"recipientId"];
-                PFQuery *userQuery = [PFUser query];
-                [userQuery whereKey:@"objectId" equalTo:friendId];
-                PFUser *user = [userQuery findObjects][0];
-                [friendArr addObject:user];
-                
-                // Fetch friendship from other side
-                PFQuery *query = [PFQuery queryWithClassName:classNameFriendship];
-                [query whereKey:@"recipientId" equalTo:self.user.objectId];
-                [query whereKey:@"requesterId" equalTo:user.objectId];
-                [otherFriendships addObject:[query findObjects][0]];
-            }
-            self.friendsArr = friendArr;
-            self.otherFriendshipArr = otherFriendships;
-            dispatch_async(dispatch_get_main_queue(), ^{
-                            NSLog(@"AYYYY");
-                            [self.tableView reloadData];
-                self.tableView.rowHeight = UITableViewAutomaticDimension;
-                        });
-        } else {
-            NSLog(@"%@",
-                  error.localizedDescription);
+    NSArray *friendships = [ParseAPIHelper fetchFriendships:self.user.objectId];
+    self.friendshipArr = friendships;
+    self.filterFriendshipArr = friendships;
```

Comment:
are they the same array? 
why do we need 2?

Reply:
I was going to implement search bar but I think I will get rid of it